src/SonarQube.Common/MsBuildPathSettings.cs

```diff
@@ -0,0 +1,56 @@
+﻿/*
+ * SonarQube Scanner for MSBuild
+ * Copyright (C) 2016-2018 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+
+namespace SonarQube.Common
+{
+    public class MsBuildPathSettings : IMsBuildPathsSettings
+    {
+        public IEnumerable<string> GetImportBeforePaths()
+        {
+            var appData = Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData);
+
+            yield return GetMsBuildImportBeforePath(appData, "15.0");
```

Comment:
You probably haven't change the logic but that looks weird to return the windows paths for non-windows OSes.

Reply:
It should be actually mapping the paths for linux:
https://ibboard.co.uk/Programming/mono-special-folders.html